solid/solidpython.py

```diff
@@ -445,21 +443,21 @@ def add_param(self, k, v):
         return self
 
     def copy(self):
-        # Provides a copy of this object and all children,
-        # but doesn't copy self.parent, meaning the new object belongs
-        # to a different tree
-        # If we're copying a scad object, we know it is an instance of
-        # a dynamically created class called self.name.
-        # Initialize an instance of that class with the same params
-        # that created self, the object being copied.
+        '''
+        Provides a copy of this object and all children,
+        but doesn't copy self.parent, meaning the new object belongs
+        to a different tree
+        Initialize an instance of this class with the same params
+        that created self, the object being copied.
+        '''
 
         # Python can't handle an '$fn' argument, while openSCAD only wants
         # '$fn'.  Swap back and forth as needed; the final renderer will
         # sort this out.
         if '$fn' in self.params:
             self.params['segments'] = self.params.pop('$fn')
 
-        other = globals()[self.name](**self.params)
+        other = type(self)(**self.params)
```

Comment:
Looking up in globals no longer works when things are across modules - but was this really desirable anyway?

This now guarantees that `type(x) == type(x.copy)`.